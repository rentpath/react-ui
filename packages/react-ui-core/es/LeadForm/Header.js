import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/inherits";
import _defineProperty from "@babel/runtime/helpers/defineProperty";

var _dec, _class, _class2, _temp;

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import themed from '@rentpath/react-themed';
import { Text } from '../Text';
var Header = (_dec = themed(/^(Modal_Header|Modal_SubHeader)/, {
  pure: true
}), _dec(_class = (_temp = _class2 =
/*#__PURE__*/
function (_PureComponent) {
  _inherits(Header, _PureComponent);

  function Header() {
    _classCallCheck(this, Header);

    return _possibleConstructorReturn(this, _getPrototypeOf(Header).apply(this, arguments));
  }

  _createClass(Header, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          theme = _this$props.theme,
          header = _this$props.header;
      return React.createElement("div", null, React.createElement(Text, {
        className: theme.Modal_Header
      }, header), this.subHeader);
    }
  }, {
    key: "subHeader",
    get: function get() {
      var subHeader = this.props.subHeader;

      if (subHeader) {
        return React.createElement(Text, {
          className: this.props.theme.Modal_SubHeader
        }, subHeader);
      }

      return null;
    }
  }]);

  return Header;
}(PureComponent), _defineProperty(_class2, "propTypes", {
  theme: PropTypes.object,
  header: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.func]),
  subHeader: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.func])
}), _defineProperty(_class2, "defaultProps", {
  theme: {},
  header: 'Contact Property'
}), _temp)) || _class);
export { Header as default };