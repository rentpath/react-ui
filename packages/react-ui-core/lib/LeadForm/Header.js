'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _dec, _class, _class2, _temp;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _reactThemed = require('react-themed');

var _reactThemed2 = _interopRequireDefault(_reactThemed);

var _Text = require('../Text');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Header = (_dec = (0, _reactThemed2.default)(/^(Modal_Header|Modal_SubHeader)/, {
  pure: true
}), _dec(_class = (_temp = _class2 = function (_PureComponent) {
  (0, _inherits3.default)(Header, _PureComponent);

  function Header() {
    (0, _classCallCheck3.default)(this, Header);
    return (0, _possibleConstructorReturn3.default)(this, (Header.__proto__ || (0, _getPrototypeOf2.default)(Header)).apply(this, arguments));
  }

  (0, _createClass3.default)(Header, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          theme = _props.theme,
          header = _props.header;


      return _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(
          _Text.Text,
          { className: theme.Modal_Header },
          header
        ),
        this.subHeader
      );
    }
  }, {
    key: 'subHeader',
    get: function get() {
      var subHeader = this.props.subHeader;


      if (subHeader) {
        return _react2.default.createElement(
          _Text.Text,
          { className: this.props.theme.Modal_SubHeader },
          subHeader
        );
      }

      return null;
    }
  }]);
  return Header;
}(_react.PureComponent), _class2.propTypes = {
  theme: _propTypes2.default.object,
  header: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.node, _propTypes2.default.func]),
  subHeader: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.node, _propTypes2.default.func])
}, _class2.defaultProps = {
  theme: {},
  header: 'Contact Property'
}, _temp)) || _class);
exports.default = Header;