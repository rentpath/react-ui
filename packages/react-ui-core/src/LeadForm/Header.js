import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import themed from 'react-themed'
import { Text } from '../Text'

@themed(/^(Modal_Header|Modal_SubHeader)/, {
  pure: true,
})

export default class Header extends PureComponent {
  static propTypes = {
    theme: PropTypes.object,
    header: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.node,
      PropTypes.func,
    ]),
    subHeader: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.node,
      PropTypes.func,
    ]),
  }

  static defaultProps = {
    theme: {},
    header: 'Contact Property',
  }

  get subHeader() {
    const { subHeader } = this.props

    if (subHeader) {
      return (
        <Text className={this.props.theme.Modal_SubHeader}>
          {subHeader}
        </Text>
      )
    }

    return null
  }

  render() {
    const {
      theme,
      header,
    } = this.props

    return (
      <div>
        <Text className={theme.Modal_Header}>
          {header}
        </Text>
        {this.subHeader}
      </div>
    )
  }
}
