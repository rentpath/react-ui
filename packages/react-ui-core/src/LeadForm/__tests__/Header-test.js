import React from 'react'
import renderer from 'react-test-renderer'
import Header from '../Header'

describe('Header', () => {
  it('renders header only if no subHeader passsed', () => {
    const snap = renderer
      .create(<Header header="Contact Property" />)
      .toJSON()
    expect(snap).toMatchSnapshot()
  })

  it('renders header and subHeader if subHeader passed', () => {
    const snap = renderer
      .create(
        <Header
          header="Contact Property"
          subHeader="To: (xxx) xxx - xxxx"
        />
      )
      .toJSON()
    expect(snap).toMatchSnapshot()
  })
})
