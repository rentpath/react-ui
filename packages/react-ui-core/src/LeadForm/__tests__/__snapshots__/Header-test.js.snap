// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders header and subHeader if subHeader passed 1`] = `
<div>
  <div
    className=""
    data-tid="text"
  >
    Contact Property
  </div>
  <div
    className=""
    data-tid="text"
  >
    To: (xxx) xxx - xxxx
  </div>
</div>
`;

exports[`Header renders header only if no subHeader passsed 1`] = `
<div>
  <div
    className=""
    data-tid="text"
  >
    Contact Property
  </div>
</div>
`;
